src/Nox.Types/Common/EnumExtensions.cs

```diff
@@ -0,0 +1,11 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Nox.Common;
+
+public static class EnumExtensions
+{
+    public static HashSet<string> ToHashSet<T>() where T : struct, Enum
+        => Enum.GetValues<T>().Select(e => nameof(e)).ToHashSet();
```

Comment:
I've rushed a bit into conclusion. We cannot use nameof(e). 

We can either dynamically create the hash set using .ToString() or explicitly provide the full list of nameof(CurrencyCode.*). The first approach does not require any code change when we add a new currency code, whereas the second one does.

These are the benchmark results after the change
![image](https://github.com/NoxOrg/Nox/assets/30667828/859ee1c1-c6a2-43cb-aec1-9eda906271cf)


Which approach should we take, @rochar?

Reply:
any of the 2 bests would be ok

Reply:
we can use the first one, not a big deal, its just done once